src/pyodide/python-entrypoint-helper.js

```diff
@@ -190,47 +199,51 @@ async function setupPackages(pyodide) {
     pyodide.FS.writeFile(
       "/lib/python3.11/site-packages/aiohttp_fetch_patch.py",
       new Uint8Array(mod.default),
-      { canOwn: true }
+      { canOwn: true },
     );
     pyodide.pyimport("aiohttp_fetch_patch");
   }
-  if (requirements.some(req => req.startsWith("fastapi"))) {
+  if (requirements.some((req) => req.startsWith("fastapi"))) {
     const mod = await import("pyodide-internal:asgi.py");
     pyodide.FS.writeFile(
       "/lib/python3.11/site-packages/asgi.py",
       new Uint8Array(mod.default),
-      { canOwn: true }
+      { canOwn: true },
     );
   }
 
   // The main module can have a `.py` extension, strip it if it exists.
   const mainName = metadata.mainModule;
-  const mainModule = mainName.endsWith(".py") ? mainName.slice(0, -3) : mainName;
+  const mainModule = mainName.endsWith(".py")
+    ? mainName.slice(0, -3)
+    : mainName;
 
   return pyodide.pyimport(mainModule);
 }
 
-let mainModule;
+let result;
 async function getMainModule() {
-  if (mainModule) {
-    return mainModule;
+  if (result) {
+    return result;
   }
   // TODO: investigate whether it is possible to run part of loadPyodide in top level scope
   // When we do it in top level scope we seem to get a broken file system.
   const pyodide = await loadPyodide();
-  mainModule = await setupPackages(pyodide);
-  return mainModule;
+  const mainModule = await setupPackages(pyodide);
+  const relaxed_call = pyodide.pyimport("relaxed_call").relaxed_call;
+  result = { mainModule, relaxed_call };
+  return result;
 }
 
 export default {
-  async fetch(request, env) {
-    const mainModule = await getMainModule();
-    return await mainModule.fetch(request);
+  async fetch(ctx, env) {
```

Comment:
Not sure we should rename `request` to `ctx` here. With `fetch(...)` handlers, the third argument is the `ExecutionContext` that, by convention, we've typically named `ctx`. Would be good to avoid confusion here.

Reply:
What about for `test`? Should the first argument always be called `request`? And presumably if `fetch` takes a third argument I should pass that along too?

Reply:
No, for `test` it should be `ctx`. I'm not a big fan of the difference but that first argument to `fetch` is a `Request` object so the name should match.

Reply:
Actually, `test`'s arguments are `test(controller, env, ctx)`. `ctx` is still the third argument. `controller` is equivalent to `request`.